flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableSink.java

```diff
@@ -161,4 +164,37 @@ private String toStagingDir(String finalDir, Configuration conf) throws IOExcept
 		fs.deleteOnExit(path);
 		return res;
 	}
+
+	@Override
+	public List<String> getPartitionFieldNames() {
+		return catalogTable.getPartitionKeys();
+	}
+
+	@Override
+	public void setStaticPartition(Map<String, String> partitions) {
+		// make it a LinkedHashMap to maintain partition column order
+		staticPartitionSpec = new LinkedHashMap<>();
+		for (String partitionCol : getPartitionFieldNames()) {
+			if (partitions.containsKey(partitionCol)) {
```

Comment:
what if it doesn't contain the key?

Reply:
It means that partition column is not contained in the static spec, and therefore it's a dynamic partition column.